Sources/SwiftDoc/Interface.swift

```diff
@@ -5,10 +5,12 @@ import struct SwiftSemantics.Protocol
 public final class Interface: Codable {
     public let imports: [Import]
     public let symbols: [Symbol]
+    public let minimumAccessLevel: AccessLevel
```

Comment:
In my point of view, adding this property on the interface mixes concerns. The setting for _creating the documentation_ should not be on a structure that describes source code. 

I'd suggest we don't include it as a property, but instead only pass the symbols that have the correct ACL. Same goes for the extension of `Report`. Alternatively we could collect and work with all symbols and then filter when we generate the documentation. But I did not fully think this through, so there might be some implications I did not consider.

But I also think @mattt is the person to decide on this.

Reply:
I agree with @Lukas-Stuehrk here. An `Interface` should represent a fixed collection of symbols and relationships. For instance, if we were to generate the public API for a library on macOS vs. Linux, we'd associate a separate `Interface` value for each platform.

As to where that determination should be made, that remains an open question. Like Lukas said, it could be done either in "model" layer (by adding a `minimumAccessLevel` parameter to `init(sourceFiles:)`) or in the "view" layer (by having each page / component filter symbols according to a `minimumAccessLevel` property). The former would be simpler, but I'm not 100% sure whether we can correctly reconcile relationships without knowledge of every symbol. For example, I know that a public class can't inherit from a private class, but I haven't thought through all of the possible interactions between protocols, types, and members.

Reply:
(@johankool Don't worry about trying to make sense of my rambling commentary here. I'll provide more actionable feedback in my review)

Reply:
Agreed. It didn’t feel quite right to put it here. Not quite sure what the right place would be though.